### ISSUE TITLE
Stage/unstage file, then immediately click it, no diff shown

### ISSUE THREAD

**Issue by roblourens** [microsoft/vscode#196325]
- Make a change in a file in a git repo
- Open SCM view, click + to stage the file, then immediately click the file's row under "staged changes"
- A diff editor opens, which says "working tree" (should say "Indexed"), and shows no changes
- Close it and click the file again, this time it opens correctly

It also does the same in reverse- if I unstage a file and immediately open it, it will open "indexed" with no changes even though the tree has updated

**Comment by lszomoru** [comment]
This occurs due to the fact that we optimistically move the resource to "Staged Changes" **before** the `git add` command completes in order to make the UI feel more snappy. You can disable the optimistic update of the Source Control view using the `git.optimisticUpdate` setting.

**Comment by roblourens** [comment]
Can the the diff view update once the index updates?

**Comment by lszomoru** [comment]
I don't think that changing the editor (tab, contents) is the ideal solution here.

**Comment by roblourens** [comment]
Why not? It would be refreshing to match reality, which is what I generally expect from an editor.

**Comment by gk-bbai** [comment]
I can confirm that setting `"git.optimisticUpdate": false` seems to fix the issue.

It's a bit odd to me that this issue doesn't have more upvotes/visibility as the default behavior breaks being able to use the diff view very often (e.g., "Undo Last Commit" could break it).
